diff -ur SDL2-2.0.5.orig/configure.in SDL2-2.0.5/configure.in
--- SDL2-2.0.5.orig/configure.in	2018-06-25 21:35:41.561999198 -0400
+++ SDL2-2.0.5/configure.in	2018-06-25 21:37:41.564633946 -0400
@@ -2885,6 +2885,18 @@
 case "$host" in
     *-*-linux*|*-*-uclinux*|*-*-gnu*|*-*-k*bsd*-gnu|*-*-bsdi*|*-*-freebsd*|*-*-dragonfly*|*-*-netbsd*|*-*-openbsd*|*-*-sysv5*|*-*-solaris*|*-*-hpux*|*-*-aix*|*-*-minix*)
         case "$host" in
+            arm-poky-linux*)
+                # Raspberry Pi
+                ARCH=linux
+                EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lbcm_host -lvchostif -ldl"
+
+                if test x$enable_video = xyes; then
+                    SOURCES="$SOURCES $srcdir/src/video/raspberry/*.c"
+                    # FIXME: confdefs? Not AC_DEFINE?
+                    $as_echo "#define SDL_VIDEO_DRIVER_RPI 1" >>confdefs.h
+                    SUMMARY_video="${SUMMARY_video} rpi"
+                fi
+                ;;
             *-raspberry-linux*)
                 # Raspberry Pi
                 ARCH=linux
diff -ur SDL2-2.0.5.orig/src/video/SDL_egl.c SDL2-2.0.5/src/video/SDL_egl.c
--- SDL2-2.0.5.orig/src/video/SDL_egl.c	2016-10-19 23:56:26.000000000 -0400
+++ SDL2-2.0.5/src/video/SDL_egl.c	2018-06-25 21:37:18.608129934 -0400
@@ -40,10 +40,10 @@
 
 #if SDL_VIDEO_DRIVER_RPI
 /* Raspbian places the OpenGL ES/EGL binaries in a non standard path */
-#define DEFAULT_EGL "/opt/vc/lib/libEGL.so"
-#define DEFAULT_OGL_ES2 "/opt/vc/lib/libGLESv2.so"
-#define DEFAULT_OGL_ES_PVR "/opt/vc/lib/libGLES_CM.so"
-#define DEFAULT_OGL_ES "/opt/vc/lib/libGLESv1_CM.so"
+#define DEFAULT_EGL "libEGL.so"
+#define DEFAULT_OGL_ES2 "libGLESv2.so"
+#define DEFAULT_OGL_ES_PVR "libGLES_CM.so"
+#define DEFAULT_OGL_ES "libGLESv1_CM.so"
 
 #elif SDL_VIDEO_DRIVER_ANDROID || SDL_VIDEO_DRIVER_VIVANTE
 /* Android */
