diff -Nur kernel-source-orig/drivers/usb/gadget/legacy/inode.c kernel-source/drivers/usb/gadget/legacy/inode.c
--- kernel-source-orig/drivers/usb/gadget/legacy/inode.c	2021-01-10 18:03:26.352594310 -0500
+++ kernel-source/drivers/usb/gadget/legacy/inode.c	2021-01-10 19:18:24.027112431 -0500
@@ -1281,7 +1281,7 @@
 
 	qual.bLength = sizeof qual;
 	qual.bDescriptorType = USB_DT_DEVICE_QUALIFIER;
-	qual.bcdUSB = cpu_to_le16 (0x0200);
+	qual.bcdUSB = desc->bcdUSB;
 
 	desc = dev->dev;
 	qual.bDeviceClass = desc->bDeviceClass;
@@ -1389,11 +1389,8 @@
 			if (value >= 0)
 				value = min (w_length, (u16) value);
 			break;
-		case USB_DT_STRING:
+		default:
 			goto unrecognized;
-
-		default:		// all others are errors
-			break;
 		}
 		break;
 
@@ -1852,7 +1849,6 @@
 			|| dev->dev->bDescriptorType != USB_DT_DEVICE
 			|| dev->dev->bNumConfigurations != 1)
 		goto fail;
-	dev->dev->bcdUSB = cpu_to_le16 (0x0200);
 
 	/* triggers gadgetfs_bind(); then we can enumerate. */
 	spin_unlock_irq (&dev->lock);
