--- git/drivers/usb/gadget/legacy/inode.c	2018-11-10 17:41:56.027719759 -0500
+++ git.new/drivers/usb/gadget/legacy/inode.c	2018-11-10 18:00:17.401165612 -0500
@@ -1284,7 +1284,7 @@
 
 	qual.bLength = sizeof qual;
 	qual.bDescriptorType = USB_DT_DEVICE_QUALIFIER;
-	qual.bcdUSB = cpu_to_le16 (0x0200);
+	qual.bcdUSB = desc->bcdUSB;
 
 	desc = dev->dev;
 	qual.bDeviceClass = desc->bDeviceClass;
@@ -1393,11 +1393,8 @@
 			if (value >= 0)
 				value = min (w_length, (u16) value);
 			break;
-		case USB_DT_STRING:
+		default:
 			goto unrecognized;
-
-		default:		// all others are errors
-			break;
 		}
 		break;
 
@@ -1856,7 +1853,6 @@
 			|| dev->dev->bNumConfigurations != 1)
 		goto fail;
 	dev->dev->bNumConfigurations = 1;
-	dev->dev->bcdUSB = cpu_to_le16 (0x0200);
 
 	/* triggers gadgetfs_bind(); then we can enumerate. */
 	spin_unlock_irq (&dev->lock);
